menas/src/main/scala/za/co/absa/enceladus/menas/auth/jwt/JwtAuthenticationFilter.scala

```diff
@@ -74,10 +75,11 @@ class JwtAuthenticationFilter @Autowired()(jwtFactory: JwtFactory) extends OnceP
   }
 
   private def parseJwtAuthorities(jwtClaims: Claims): util.List[SimpleGrantedAuthority] = {
-    val groups = Option(jwtClaims.get(GroupsKey, classOf[util.List[String]]))
-      .getOrElse(new util.ArrayList[String]()).asScala
+    val groups: Seq[String] = Option(jwtClaims.get(RolesKey, classOf[util.List[String]]))
+      .getOrElse(new util.ArrayList[String]())
+      .asScala
 
-    groups.map(k => new SimpleGrantedAuthority(k)).asJava
+    AuthConstants.filterByRolesRegex(groups).map(k => new SimpleGrantedAuthority(k)).asJava
```

Comment:
Why are filtering the roles here again. They come from jwt, don't they?

Reply:
Great catch! They do come in the JWT token, yes. But if you redeploy and have an old token or something (have a single token valid for multiple apps) then the roles are not filtered later. So I thought it was ok to do it even here.

Reply:
Fixed. Only in `MenasAuthenticationSuccessHandler`